python/lsst/resources/http.py

```diff
@@ -531,49 +580,269 @@ def transfer_from(
         if transfer == "auto":
             transfer = self.transferDefault
 
-        if isinstance(src, type(self)):
+        # We can use webDAV 'COPY' or 'MOVE' if both the current and source
+        # resources are located in the same server.
+        if isinstance(src, type(self)) and self.root_uri() == src.root_uri():
             # Only available on WebDAV backends.
             if not self.is_webdav_endpoint:
-                raise NotImplementedError("Endpoint does not implement WebDAV functionality")
+                raise NotImplementedError(
```

Comment:
Is it possible to try falling back to the download/upload approach just in case PUT is allowed even if it's not WebDAV?

Reply:
Fixed. `transfer_from` using plain HTTP servers now use a download-then-upload operation.